docs/security-o2k.md

```diff
@@ -0,0 +1,119 @@
+# OpenAPI2Kong support for security directives
+
+This document documents the use of `security` directives within the `deck` subcommand `openapi2kong`. There are 2 sections;
+
+1. [OpenAPI 3 security directives explained](#openapi-3-security-directives-explained)
+2. [Conversion logic to Kong plugins](#conversion-logic-to-kong-plugins)
+
+---
+
+# Openapi 3 security directives explained
+
+Within OpenAPI the security directives can be specified on the document root (the `OpenAPI` object). It can also be specified on the `Operation` object, in which case it will override the document level one.
+Specifically; it cannot be specified on the `path` object level, the level in between document and operation.
+
+## Format
+
+The `security` property is an array of `Security Requirement` objects.
+
+## Behaviour
+
+- The requirements listed are a logical `OR`; passing any one requirement listed is enough to be allowed access.
+- The default value is `[]`; an empty array
+- An empty array allows access without authentication
+- An empty array (on the `Operation` level) can override a non-empty array (document level)
+
+## Security Requirement
+
+A `security requirement` object lists (as properties) `security scheme` objects that MUST all be satisfied to allow access.
```

Comment:
The OAS document does a good job of describing the AND / OR relationship here.  https://swagger.io/docs/specification/authentication/. I'm worried this documentation confuses matters as it's not as clear as what is written there.

Reply:
Agreed.

thx for that link. Never seen before. The OAS spec itself is very terse on this subject. Hence this doc is for now mostly intended as a "thinking" doc. All OAS specifics we should probably remove (or add that link) before merging.